File: 프로그래머스/0/120866. 안전지대/안전지대.java
class Solution {
    public int solution(int[][] board) {
        int answer = 0;
        
        for (int i=0; i<board[0].length; i++) {
            for (int j=0; j<board[0].length; j++) {
                if (board[i][j] == 0) {
                    if (i > 0) {
                        if (j > 0) {
                            if (board[i-1][j-1] == 1) continue;
                        }

                        if (board[i-1][j] == 1) continue;

                        if (j < board[0].length-1) {
                            if (board[i-1][j+1] == 1) continue;
                        }
                    }
                    
                    if (j > 0) {
                        if (board[i][j-1] == 1) continue;    
                    }
                    if (j < board[0].length-1) {
                        if (board[i][j+1] == 1) continue;
                    }
                    
                    
                    if (i < board[0].length-1) {
                        if (j > 0) {
                            if (board[i+1][j-1] == 1) continue;
                        }
                        
                        if (board[i+1][j] == 1) continue;
                            
                        if (j < board[0].length-1) {
                            if (board[i+1][j+1] == 1) continue;    
                        }
                    } 
                    answer++;
                }
            }
        }
        
        return answer;
    }
}

File: 프로그래머스/0/120866. 안전지대/안전지대.py
def solution(board):
    answer = 0
    
    for i, w in enumerate(board):
        for j, h in enumerate(w):
            if board[i][j] == 0:
                if i > 0:
                    if j > 0:
                        if board[i-1][j-1] == 1: continue
                    if board[i-1][j] == 1: continue
                    if j < len(board)-1:
                        if board[i-1][j+1] == 1: continue
                if j > 0:
                    if board[i][j-1] == 1: continue
                if j < len(board)-1:
                    if board[i][j+1] == 1: continue

                if i < len(board)-1:
                    if j > 0:
                        if board[i+1][j-1] == 1: continue
                    if board[i+1][j] == 1: continue
                    if j < len(board)-1:
                        if board[i+1][j+1] == 1: continue
                answer += 1
    
    return answer